### PR TITLE
Enable mixed CUDA/ROCm benchmarks

### PR DESCRIPTION
**Pull request by upsj**
This PR moves the CUDA/HIP benchmark linops and timers into separate libraries, enabling their simultaneous compilation.

### CODE REVIEW

File: benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -87,16 +87,43 @@ endfunction(ginkgo_add_single_benchmark_executable)
...
+        "${name}_dcomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_DOUBLE_COMPLEX_PRECISION" "z" ${ARGN})
     ginkgo_add_single_benchmark_executable(
-        "${name}_scomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_SINGLE_COMPLEX_PRECISION" ${ARGN})
+        "${name}_scomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_SINGLE_COMPLEX_PRECISION" "c" ${ARGN})
 endfunction(ginkgo_add_typed_benchmark_executables)
 
 
+if (GINKGO_BUILD_CUDA)
+    enable_language(CUDA)
+    ginkgo_benchmark_cusp_linops(d GKO_BENCHMARK_USE_DOUBLE_PRECISION)
```
Comment (tcojean) [inline benchmark/CMakeLists.txt:102]: I'm not too much of a fan of the current setup for multiprecision in general, I believe it should be shortened. But that's probably better in another PR.

**Review by tcojean — APPROVED** [review]
LGTM.

File: benchmark/utils/cuda_linops.cu [hunk 2]
```diff
@@ -685,35 +694,58 @@ class CuspGenericCoo
...
+        // defined(__CYGWIN__)))))
 
 
-#endif  // GKO_BENCHMARK_UTILS_CUDA_LINOPS_HPP_
+#if defined(CUDA_VERSION) && (CUDA_VERSION < 11000)
+IMPL_CREATE_SPARSELIB_LINOP(
+    cusp_coo, detail::CuspHybrid<etype, itype, CUSPARSE_HYB_PARTITION_USER, 0>)
+IMPL_CREATE_SPARSELIB_LINOP(
+    cusp_ell, detail::CuspHybrid<etype, itype, CUSPARSE_HYB_PARTITION_MAX, 0>)
+IMPL_CREATE_SPARSELIB_LINOP(cusp_hybrid, detail::CuspHybrid<etype, itype>)
```
[the remark this answers is not in the document]
Comment (upsj) [reply benchmark/utils/cuda_linops.cu:746]: seems to be fine

File: benchmark/utils/timer_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+
+#include <chrono>
+#include <memory>
+
+
+/**
+ * Timer stores the timing information
+ */
+class Timer {
```
[the remark this answers is not in the document]
Comment (upsj) [reply benchmark/utils/timer_impl.hpp:47]: I needed a header that does not include gflags, so I needed to separate the two parts.

**Pull request merged**